### ISSUE TITLE
bond0 creation broken on some device types

### ISSUE THREAD

**Issue by hh** [siderolabs/talos#8443]
```[talos] restarting platform network config {"component": "controller-runtime", "controller": "network.PlatformConfigController", "interval": "1m29.590345908s", "error": "encountered multiple bonds. this is unexpected in the equinix metal platform"}```

> Talosctl dames 
```
147.75.85.111: user: warning: [2024-03-14T23:11:40.526566847Z]: [talos] fetching equinix network config from: "https://metadata.platformequinix.com/metadata"
147.75.85.111: user: warning: [2024-03-14T23:11:40.545012847Z]: [talos] restarting platform network config {"component": "controller-runtime", "controller": "network.PlatformConfigController", "interval": "52.685921926s", "error": "encountered multiple bonds. this is unexpected in the equinix metal platform"}
147.75.85.111: user: warning: [2024-03-14T23:11:47.547084847Z]: [talos] request/renew failed {"component": "controller-runtime", "controller": "network.OperatorSpecController", "operator": "dhcp4", "error": "unable to receive an offer: got an error while the discovery request: no matching response packet received", "link": "ens6f3"}
147.75.85.111: user: warning: [2024-03-14T23:11:47.963081847Z]: [talos] request/renew failed {"component": "controller-runtime", "controller": "network.OperatorSpecController", "operator": "dhcp4", "error": "unable to receive an offer: got an error while the discovery request: no matching response packet received", "link": "ens6f2"}
147.75.85.111: user: warning: [2024-03-14T23:11:48.047061847Z]: [talos] request/renew failed {"component": "controller-runtime", "controller": "network.OperatorSpecController", "operator": "dhcp4", "error": "unable to receive an offer: got an error while the discovery request: no matching response packet received", "link": "ens6f1"}
147.75.85.111: user: warning: [2024-03-14T23:12:27.586730847Z]: [talos] request/renew failed {"component": "controller-runtime", "controller": "network.OperatorSpecController", "operator": "dhcp4", "error": "unable to receive an offer: got an error while the discovery request: no matching response packet received", "link": "ens6f3"}
147.75.85.111: user: warning: [2024-03-14T23:12:27.986702847Z]: [talos] request/renew failed {"component": "controller-runtime", "controller": "network.OperatorSpecController", "operator": "dhcp4", "error": "unable to receive an offer: got an error while the discovery request: no matching response packet received", "link": "ens6f2"}
147.75.85.111: user: warning: [2024-03-14T23:12:28.078713847Z]: [talos] request/renew failed {"component": "controller-runtime", "controller": "network.OperatorSpecController", "operator": "dhcp4", "error": "unable to receive an offer: got an error while the discovery request: no matching response packet received", "link": "ens6f1"}
147.75.85.111: user: warning: [2024-03-14T23:12:32.432922847Z]: [talos] controller failed {"component": "controller-runtime", "controller": "network.RouteSpecController", "error": "2 errors occurred:\n\t* error adding route: netlink receive: network is unreachable, message {Family:2 DstLength:32 SrcLength:0 Tos:0 Table:0 Protocol:4 Scope:0 Type:1 Flags:0 Attributes:{Dst:169.254.255.1 Src:<nil> Gateway:10.12.190.129 OutIface:8 Priority:1024 Table:254 Mark:0 Pref:<nil> Expires:<nil> Metrics:<nil> Multipath:[]}}\n\t* error adding route: netlink receive: network is unreachable, message {Family:2 DstLength:32 SrcLength:0 Tos:0 Table:0 Protocol:4 Scope:0 Type:1 Flags:0 Attributes:{Dst:169.254.255.2 Src:<nil> Gateway:10.12.190.129 OutIface:8 Priority:1024 Table:254 Mark:0 Pref:<nil> Expires:<nil> Metrics:<nil> Multipath:[]}}\n\n"}
147.75.85.111: user: warning: [2024-03-14T23:12:33.231577847Z]: [talos] fetching equinix network config from: "https://metadata.platformequinix.com/metadata"
147.75.85.111: user: warning: [2024-03-14T23:12:33.245822847Z]: [talos] restarting platform network config {"component": "controller-runtime", "controller": "network.PlatformConfigController", "interval": "1m29.590/q345908s", "error": "encountered multiple bonds. this is unexpected in the equinix metal platform"}q
```

Sent support.zip directly, but can attach to issue if needed.

**Comment by smira** [comment]
Any chance you can capture the https://metadata.platformequinix.com/metadata output for that instance?

**Comment by hh** [comment]
I noticed that these were closed:
- https://github.com/siderolabs/talos/issues/8267#event-12199873824
- https://github.com/siderolabs/talos/issues/8411#event-12199873941

Just wanting to check if the bond creation, BPG routes, etc were part of the process.
I can test sometime, but want to make sure I use an image that has all the changes merged.

**Comment by frezbo** [comment]
> I noticed that these were closed:
> 
>     * [Pull BGP peering info out of metadata for equinixMetal platform #8267 (comment)](https://github.com/siderolabs/talos/issues/8267#event-12199873824)
> 
>     * [Equinix Metal retired Talos February 16th #8411 (comment)](https://github.com/siderolabs/talos/issues/8411#event-12199873941)
> 
> 
> Just wanting to check if the bond creation, BPG routes, etc were part of the process. I can test sometime, but want to make sure I use an image that has all the changes merged.

it should be in the next beta

**Comment by smira** [comment]
In fact, this issue should be resolved by now (in 1.7-beta.0 as it gets cut), but if you see it once again, please post with the metadata or a way to reproduce.
